src/piadm.sh

```diff
@@ -195,6 +196,102 @@ DEFAULT_URL_PREFIX=https://us-central.manta.mnx.io/Joyent_Dev/public/SmartOS/
 # Default path for piadm's configuration
 PIADM_CONF=/var/piadm/piadm.conf
 
+# fetch_md5sum
```

Comment:
As I mentioned earlier, we *do* use MD5 all over for this right now.  On the off chance we move to some other hash (e.g. SHA256) in the future, do we want to be algorithm-agile here?

I think the function name (fetch_md5sum) should be independent (fetch_sum).  I think the environment names should be independent (see above).  I think internal variable names should be independent too.